### PR TITLE
chore: fix errors

### PR DESCRIPTION
**Pull request by ToastedDev**
another self explanatory pr, just fixes most of the typescript and eslint errors
some of them I had to `// eslint-disable-next-line` because there's no other fix for it

### CODE REVIEW

File: bot/index.ts [hunk 1]
```diff
@@ -22,14 +22,15 @@ const rest = new REST().setToken(discordToken)
...
-const data: any = await rest.put(
+const data = await rest.put(
 	Routes.applicationCommands(getAppId.id),
 	{
 		body: [...commandsMap.values()].map((a) => {
 			return a.data;
 		}),
 	},
-);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (GalvinPython) [inline bot/index.ts:32]: need to fix at some point

File: api/index.ts [hunk 2]
```diff
@@ -303,6 +303,7 @@ app.get("/leaderboard/:guild", async (req, res) => {
 
 app.get("/", async (_req, res) => {
 	// TODO: handle error
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (GalvinPython) [inline api/index.ts:306]: need to fix at some point

File: bot/index.ts (hunk 1 above)
Comment (ToastedDev) [reply bot/index.ts:32]: yeah there needs to be a real type for it, i'll try to see if there's a type for it in d.js or @d.js/rest

File: api/index.ts (hunk 2 above)
Comment (ToastedDev) [reply api/index.ts:306]: could maybe return 500 if it errors?

**Review by GalvinPython — APPROVED** [review]
thanks for fixing the eslint issues. Don't want to have the ignore comments, but that's for another time

File: api/index.ts (hunk 2 above)
Comment (GalvinPython) [reply api/index.ts:306]: i'll handle the error page stuff 👍

**Pull request merged**